### PR TITLE
Scoring Class, FileIO and Exceptions, Row highlighting and gameover

### PR DESCRIPTION
**Pull request by LukasK-roboyor**
- Added TetrisScore class, which handles everything related to the score (i.e updating, drawing, saving)
- Moved scoring methods and variables from Game.cpp to the TetrisScore class

- Added class for FileIO, which can write/read from text files
- Highscores can now be read and written to text file (Highscores.txt) 
- Added exceptions

- Completed rows now turn white for the duration of 1 tick before they get deleted

- The playfield turns completely white, if gameover is reached

### CODE REVIEW

File: src/Exceptions/BasicException.h [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#pragma once
+#include <string>
+class BasicException
```
Comment (Maze7) [inline src/Exceptions/BasicException.h]: maybe inheritance would sensible

File: src/Exceptions/FileNotFoundException.h [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#pragma once
+#include "BasicException.h"
+
+class FileNotFoundException : public BasicException
```
Comment (Maze7) [inline src/Exceptions/FileNotFoundException.h]: maybe a namespace make sense

File: src/FileIO/FileIO.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
+#include "FileIO.h"
+
+FileIO::FileIO()
```
Comment (Maze7) [inline src/FileIO/FileIO.cpp]: put declaration to header if budy is empty

File: src/GameCollection/Tetris/TetrisScore.h [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+{
+	struct Highscore
+	{
+		Highscore(int score, int level, int lineCount)
+			: score(score)
+			, level(level)
+			, lineCount(lineCount)
+		{}
+
+		int score;
```
Comment (Maze7) [inline src/GameCollection/Tetris/TetrisScore.h]: public variables are discouraged

File: src/FileIO/FileIO.cpp (hunk 3 above)
Comment (LukasK-roboyor) [reply src/FileIO/FileIO.cpp]: changed in 55bf794e2ca2b15632e0d2da5bece07c1841feb3

File: src/GameCollection/Tetris/TetrisScore.h (hunk 4 above)
Comment (LukasK-roboyor) [reply src/GameCollection/Tetris/TetrisScore.h]: changed in 40de18d3aac24ff04101d476c39635118616c5d8

File: src/Exceptions/FileNotFoundException.h (hunk 2 above)
Comment (LukasK-roboyor) [reply src/Exceptions/FileNotFoundException.h]: see 879138ab3edd4c38f61779c27c48ff054038af53 and 2ce170893bb23ccb17328fd6f0d44d704e553ffc

File: src/Exceptions/BasicException.h (hunk 1 above)
Comment (LukasK-roboyor) [reply src/Exceptions/BasicException.h]: see 879138a and 2ce1708

File: src/Exceptions/FileNotFoundException.cpp [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#include "FileNotFoundException.h"
```
[the remark this answers is not in the document]
Comment (LukasK-roboyor) [reply src/Exceptions/FileNotFoundException.cpp]: see 879138a and 2ce1708

File: src/Exceptions/BasicException.cpp [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#include "BasicException.h"
```
[the remark this answers is not in the document]
Comment (LukasK-roboyor) [reply src/Exceptions/BasicException.cpp]: see 879138a and 2ce1708

File: src/GameCollection/Tetris/Game.cpp [hunk 7]
```diff
@@ -269,40 +241,23 @@ bool TetrisGame::Game::isPosValid(Tetromino* tetromino)
...
 */
-void TetrisGame::Game::updateScoreSystem(uint completedRowCount)
+void TetrisGame::Game::checkForGameOver()
 {
-	// Update the line count
-	m_lineCount += completedRowCount;
-
-	// Increase the score depending on rows completed
-	switch (completedRowCount)
+	for (int x = 0; x < 10; x++)
```
Comment (Maze7) [inline src/GameCollection/Tetris/Game.cpp]: maybe use the defined playfield size out of Playfield class instead of hardcode the number

**Pull request merged**